viewbuilder-rx2/src/main/java/com/squareup/viewbuilder/StackScreen.kt

```diff
@@ -17,10 +17,22 @@ package com.squareup.viewbuilder
 
 import io.reactivex.Observable
 
-data class ViewStackScreen<T : Any>(
+/**
+ * Wraps screens that may be shown in series, drill down or wizard style.
+ * Typically these are the leaves of composite UI structures. That is, it's
+ * probably a mistake if you find yourself creating, say, a
+ * `StackScreen<MainAndModalScreen<*, *>>`.
+ */
+data class StackScreen<out T : Any>(
   val wrapped: T,
   private val keyExtension: String = ""
 ) {
+  init {
+    require(wrapped !is StackScreen<*>) {
```

Comment:
@zach-klippenstein kind of Nanny-ish, but I already made this mistake a couple of times today.

Reply:
Seems fine just document as well.